src/containers/Layout/ScrollToTop/index.tsx

```diff
@@ -0,0 +1,39 @@
+import { ButtonWrapper, Icon } from 'components';
+import { useEffect, useState } from 'react';
+import { cn } from 'utilities';
+
+const ScrollToTop = () => {
+  const [isVisible, setIsVisible] = useState(false);
+  const viewportHeight = window.innerHeight;
+  const scrollElem = document.getElementById('page-container');
+
+  useEffect(() => {
+    const toggleVisibility = () => {
+      if (scrollElem?.scrollTop && scrollElem.scrollTop > viewportHeight) {
+        setIsVisible(true);
+      } else {
+        setIsVisible(false);
+      }
+    };
+
+    scrollElem?.addEventListener('scroll', toggleVisibility);
+
+    return () => scrollElem?.removeEventListener('scroll', toggleVisibility);
+  }, [scrollElem]);
```

Comment:
I thought this would solve the issue with the effect not working, but unfortunately it's still there. The issue is that `scrollElem` is `null` when the component initially renders, and because it never re-renders then the effect does not work.

An alternative solution to the problem would be to detect scroll events on the page container by passing and `onScroll` prop to it. You could then use a zustand store (see example [here](https://github.com/VenusProtocol/venus-protocol-interface/blob/962b991337414b940c39f15a0e4247dd8f6b6b28/src/pages/Dashboard/PrimePromotionalBanner/store.ts)) to store whether the scroll to top button should show or not and listen to changes to that store in scroll to top button itself.
Let me know your thoughts.